src/Compilers/Core/Portable/Syntax/InternalSyntax/SyntaxDiagnosticInfoList.cs

```diff
@@ -47,14 +50,14 @@ internal NodeIteration(GreenNode node)
                 }
             }
 
-            private NodeIteration[] _stack;
+            private NodeIteration[]? _stack;
             private int _count;
 
             public DiagnosticInfo Current { get; private set; }
 
             internal Enumerator(GreenNode node)
             {
-                Current = null;
+                Current = null!;
                 _stack = null;
```

Comment:
💭 Seems like this could be initialized to `Array.Empty<NodeIteration>()`. Would require `Push` be updated to handle a length of 0 though.

Reply:
Given this is a `struct` we'd still have to consider the case where `_stack` was `null`. That or convince ourselves it was a non-issue. 

---
In reply to: [363548332](https://github.com/dotnet/roslyn/pull/40767#discussion_r363548332) [](ancestors = 363548332)